scenarios/pravega-flink-connector-sql-samples/src/main/java/io/pravega/connectors/nytaxi/PopularDestinationQuery.java

```diff
@@ -0,0 +1,87 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ */
+package io.pravega.connectors.nytaxi;
+
+import io.pravega.client.stream.Stream;
+import io.pravega.connectors.flink.FlinkPravegaJsonTableSource;
+import io.pravega.connectors.nytaxi.common.TripRecord;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
+import org.apache.flink.table.api.Table;
+import org.apache.flink.table.api.TableEnvironment;
+import org.apache.flink.table.api.TableSchema;
+import org.apache.flink.table.api.java.StreamTableEnvironment;
+import org.apache.flink.table.sources.tsextractors.ExistingField;
+import org.apache.flink.table.sources.wmstrategies.BoundedOutOfOrderTimestamps;
+import org.apache.flink.types.Row;
+
+/**
+ * Identify the popular destination based on total trips that exceeds a given threshold for specific time window.
+ */
+
+@Slf4j
+public class PopularDestinationQuery extends AbstractHandler {
+
+    public PopularDestinationQuery (String ... args) {
+        super(args);
+    }
+
+    @Override
+    public void handleRequest() {
+
+        TableSchema tableSchema = TripRecord.getTableSchema();
+
+        FlinkPravegaJsonTableSource source = FlinkPravegaJsonTableSource.builder()
+                .forStream(Stream.of(getScope(), getStream()).getScopedName())
+                .withPravegaConfig(getPravegaConfig())
+                .failOnMissingField(true)
+                .withRowtimeAttribute("pickupTime",
+                        new ExistingField("pickupTime"),
+                        new BoundedOutOfOrderTimestamps(30000L))
+                .withSchema(tableSchema)
+                .withReaderGroupScope(getScope())
+                .build();
+
+
+        StreamExecutionEnvironment env = getStreamExecutionEnvironment();
+
+        // create a TableEnvironment
+        StreamTableEnvironment tEnv = TableEnvironment.getTableEnvironment(env);
+        tEnv.registerTableSource("TaxiRide", source);
+
+        String query =
+                "SELECT " +
+                        "destLocationId, wstart, wend, cnt " +
+                        "FROM " +
+                        "(SELECT " +
+                        "destLocationId, " +
+                        "HOP_START(pickupTime, INTERVAL '5' MINUTE, INTERVAL '15' MINUTE) AS wstart, " +
+                        "HOP_END(pickupTime, INTERVAL '5' MINUTE, INTERVAL '15' MINUTE) AS wend, " +
+                        "COUNT(destLocationId) AS cnt " +
+                        "FROM " +
+                        "(SELECT " +
+                        "pickupTime, " +
+                        "destLocationId " +
+                        "FROM TaxiRide) " +
+                        "GROUP BY destLocationId, HOP(pickupTime, INTERVAL '5' MINUTE, INTERVAL '15' MINUTE)) " +
+                        "WHERE cnt > " + getLimit();
```

Comment:
I'm not going to block the PR on this comment, but it would be much cooler if instead of having hard threshold, that we took like the heavy hitter of the distribution of counts. I suspect that the distribution is going to be heavy-tailed, so we can take like the top-n destinations.